src/core/experimental/memory/memcpy.d

```diff
@@ -0,0 +1,431 @@
+/**
+ * Pure D replacement of the C Standard Library memcpy().
+ * There is an idiomatic-D interface, memcpy(), which is split into 3 overloads.
+ * One taking static types, one dynamic arrays and one static arrays.
+ * Also, there is available a C-like interface, the Dmemcpy() (which is named Dmemcpy
+ * for disambiguation with the C memcpy() which has the exact same interface) that
+ * is the classic (void*, void*, size_t) interface.
+ * Source: $(DRUNTIMESRC core/experimental/memory/memcpy.d)
+ */
+module core.experimental.memory.memcpy;
+
+import core.internal.traits : isArray;
+
+/* Static Types
+ */
+// NOTE(stefanos): Previously, there was more sophisticated code
+// for static types. But the rationale of removing it is that
+// the compiler knows better how to optimize static types.
+pragma(inline, true)
+void memcpy(T)(ref T dst, ref const T src)
+if (!isArray!T)
+{
+    dst = src;
+}
+
+/* Dynamic Arrays
+ */
+void memcpy(T)(ref T[] dst, ref const T[] src)
+{
+    assert(dst.length == src.length);
+    void* d = cast(void*) dst.ptr;
+    const(void)* s = cast(const(void)*) src.ptr;
+    size_t n = dst.length * typeof(dst[0]).sizeof;
+    // Assume that there is no overlap.
+    pragma(inline, true);
+    Dmemcpy(d, s, n);
+}
+
+/* Static Arrays
+ */
+void memcpy(T, size_t len)(ref T[len] dst, ref const T[len] src)
+{
+    T[] d = dst[0 .. $];
+    const T[] s = src[0 .. $];
+    memcpy(d, s);
+}
+
+/** Tests
+  */
+
+/* Basic features tests
+ */
+unittest
+{
+    real a = 1.2;
+    real b;
+    memcpy(b, a);
+    assert(b == 1.2);
+}
+///
+unittest
+{
+    const float[3] a = [1.2, 3.4, 5.8];
+    float[3] b;
+    memcpy(b, a);
+    assert(b[0] == 1.2f);
+    assert(b[1] == 3.4f);
+    assert(b[2] == 5.8f);
+}
+
+/* More sophisticated test suite
+ */
+version (unittest)
+{
+    import std.random;
```

Comment:
No Phobos in druntime.

Reply:
Hey, we saw it the same time. Just updated with non-random values since it doesn't really matter.